services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/IntentionalRerouteSpec.groovy

```diff
@@ -117,7 +125,46 @@ class IntentionalRerouteSpec extends BaseSpecification {
         and: "Remove the flow, restore bandwidths on ISLs, reset costs"
         Wrappers.wait(WAIT_OFFSET) { assert northbound.getFlowStatus(flow.id).status == FlowState.UP }
         flowHelper.deleteFlow(flow.id)
-        [thinIsl, islUtils.reverseIsl(thinIsl)].each { database.revertIslBandwidth(it) }
+        [thinIsl, thinIsl.reversed].each { database.revertIslBandwidth(it) }
+    }
+
+    def "Intentional flow reroute is not causing any packet loss"() {
+        given: "An unmetered flow going through unpreferable path(reroute potential)"
+        //will be available on virtual as soon as we get the latest iperf installed in lab-service images
+        requireProfiles("hardware")
+        def flow = flowHelper.randomFlow(topology.activeTraffGens[0].switchConnected,
+                topology.activeTraffGens[1].switchConnected)
+        flow.maximumBandwidth = 0
+        flow.ignoreBandwidth = true
+        flowHelper.addFlow(flow)
+        def path = northbound.getFlowPath(flow.id)
+        pathHelper.makePathUnpreferable(path)
+
+        when: "Start traffic examination"
+        def traffExam = traffExamProvider.get()
+        def bw = 100000 // 100 Mbps
+        def exam = new FlowTrafficExamBuilder(topology, traffExam).buildBidirectionalExam(flow, bw)
+        [exam.forward, exam.reverse].each { direction ->
+            def resources = traffExam.startExam(direction, true)
+            direction.setResources(resources)
+        }
+
+        and: "While traffic flow is active, request a flow reroute"
+        [exam.forward, exam.reverse].each { assert !traffExam.isFinished(it) }
+        northbound.rerouteFlow(flow.id)
+
+        then: "Flow is rerouted"
+        northbound.getFlowPath(flow.id) != path
+
+        and: "Traffic examination result shows acceptable packet loss percentage"
+        def examReports = [exam.forward, exam.reverse].collect { traffExam.waitExam(it) }
+        examReports.each {
+            //Minor packet loss is considered a measurement error and happens regardless of reroute
+            assert it.consumerReport.lostPercent < 1
+        }
+
+        and: "Remove flow"
+        flowHelper.deleteFlow(flow.id)
```

Comment:
It looks like we should also to delete created link props.

Reply:
there is a `cleanup` method in this test
```
    def cleanup() {
        northbound.deleteLinkProps(northbound.getAllLinkProps())
        database.resetCosts()
    }
```

Reply:
Yep, didn't notice it.